core/src/main/java/org/bitcoinj/core/ECKey.java

```diff
@@ -186,16 +191,24 @@ protected ECKey(@Nullable BigInteger priv, ECPoint pub) {
      * Utility for compressing an elliptic curve point. Returns the same point if it's already compressed.
      * See the ECKey class docs for a discussion of point compression.
      */
-    public static ECPoint compressPoint(ECPoint uncompressed) {
-        return CURVE.getCurve().decodePoint(uncompressed.getEncoded(true));
+    public static ECPoint compressPoint(ECPoint point) {
+        return point.isCompressed() ? point : CURVE.getCurve().decodePoint(point.getEncoded(true));
```

Comment:
The deprecation attracts our attention to the fact that ECPoint will not maintain the compression status in the future, and that we should maintain it ourselves.  The existing code manages to avoid the warning through a technicality (i.e. using an encode/decode cycle).  We actually want that warning, to remind us to change the way we handle the compression status.